src/anomalib/data/predict.py

```diff
@@ -39,13 +39,14 @@ def __len__(self) -> int:
         """Get the number of images in the given path."""
         return len(self.image_filenames)
 
-    def __getitem__(self, index: int) -> dict[str, Any]:
+    def __getitem__(self, index: int) -> ImageItem:
         """Get the image based on the `index`."""
         image_filename = self.image_filenames[index]
         image = read_image(image_filename, as_tensor=True)
         if self.transform:
             image = self.transform(image)
-        pre_processed = {"image": image}
-        pre_processed["image_path"] = str(image_filename)
 
-        return pre_processed
+        return ImageItem(
+            image=image,
+            image_path=str(image_filename),
```

Comment:
Path instances cannot be collated by torch's `default_collate` function, which is used to collate the individual fields of the batches. The `ImageBatch` can accept Path instances, but the validation method will cast it to string.

Reply:
> The ImageBatch can accept Path instances, but the validation method will cast it to string.

Yeah, that's what I meant. I think it would be nice if `ImageBatch` cast it to str under the hood.